### PR TITLE
Simplified Contributions and Automated development setup using Gitpod

### PR DESCRIPTION
**Pull request by anudeepreddy**
This PR adds a Gitpod configuration file which will automate setting up the developer environment and make it easy for contributors to directly start working without having to worry about installing the dependencies.

image

### Added .gitpod.yml
This file automates installing the required dependencies, build and start the application. The preview is launched once the build has finished, you can go to the open ports tab and launch the application in the browser.

image

Click on this link to give it a try : https://gitpod.io/#https://github.com/anudeepreddy/carbon

It works great and it would be much better if someone from the team has a closer look at this and provide their reviews. I hope this helps the team and the dev community.

### CODE REVIEW

**Comment by anudeepreddy** [comment]
Hey @mfix22,
Rebased on top of master, please do have a look when you find time.

File: .gitpod.yml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+ports:
+- port: 3000
+  onOpen: open-preview
+tasks:
+- init: yarn
+  command: yarn dev
+- openMode: split-right
+  command: echo -e "If you made any changes to this project enter your Github username below and add yourself as a contributor\n" &&
```
Comment (mfix22) [inline .gitpod.yml]: If you just run `yarn contrib:add` on its own, it will ask the user for their username and contributions in a really pleasant manner.

Comment (anudeepreddy) [reply .gitpod.yml]: Thanks for telling me the better way to do it. I made the required changes, I hope you are good to go.

File: .gitpod.yml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+ports:
+- port: 3000
+  onOpen: open-preview
+tasks:
+- init: yarn && gp sync-done yarn
```
Comment (mfix22) [inline .gitpod.yml]: @anudeepreddy what is `gp sync-done yarn`? This file was very understandable before, but now it seems like added maintenance burden. I want to make sure this is added actual developer value without increasing maintenance burden

Comment (mfix22) [reply .gitpod.yml]: Has anyone else in the Carbon community used Gitpod and can vouch for its benefits?

Comment (anudeepreddy) [reply .gitpod.yml]: The file opens two different terminals when the workspace is launched for the first time. One to install the dependencies and the other to ask for their contributions. So before we can run `yarn contrib:add` the first terminal should complete installation of all the dependencies. So `gp sync-done yarn` will notify terminals waiting that the job is done. While `gp sync-await yarn` will wait until it is signalled that the job yarn is done.

You can run the workspace yourself by clicking on the link in this PR and in the terminal type `gp` and that will give you the usage of each option.
I will be happy to answer your questions if you have any.

Comment (mfix22) [reply .gitpod.yml]: Would it be possible to just leave the contributions stuff as is? That would simplify this file a lot.

For context, we already have a bot that handles it (@all-contributors)

Comment (anudeepreddy) [reply .gitpod.yml]: Yeah we can do that 🙂

Comment (anudeepreddy) [reply .gitpod.yml]: I made the changes. Now I hope this PR is ready to be merged

**Pull request merged**